packages/cli-platform-android/src/commands/runAndroid/index.ts

```diff
@@ -43,32 +37,8 @@ type AndroidProject = NonNullable<Config['project']['android']>;
 async function runAndroid(_argv: Array<string>, config: Config, args: Flags) {
   const androidProject = getAndroidProject(config);
 
-  if (!args.packager) {
-    return buildAndRun(args, androidProject);
-  }
-
-  return isPackagerRunning(args.port).then((result: string) => {
-    if (result === 'running') {
-      logger.info('JS server already running.');
-    } else if (result === 'unrecognized') {
-      logger.warn('JS server not recognized, continuing with build...');
-    } else {
-      // result == 'not_running'
-      logger.info('Starting JS server...');
-      try {
-        startServerInNewWindow(
-          args.port,
-          args.terminal,
-          config.reactNativePath,
-        );
-      } catch (error) {
-        logger.warn(
-          `Failed to automatically start the packager server. Please run "react-native start" manually. Error details: ${error.message}`,
-        );
-      }
-    }
-    return buildAndRun(args, androidProject);
-  });
+  await checkPackager(args, config);
```

Comment:
how about renaming this to `runPackager`? I thought about `runPackagerIfNeeded`, but I figured it's up to the "run packager" to determine whether to bail or not